sdk/cosmos/azure-cosmos/azure/cosmos/_synchronized_request.py

```diff
@@ -167,20 +169,31 @@ def _Request(global_endpoint_manager, request_params, connection_policy, pipelin
     return result, headers
 
 
+def _replace_url_prefix(original_url, new_prefix):
```

Comment:
I wonder if this call can hit some weird scenarios - do we have a fallback in such cases?

Reply:
every url that reaches this code has already been created or handled by the SDK previously, so even if we were to hit a weird scenario that would mean something earlier on in the pipeline is returning a wrongly formatted url - which would itself be some sort of regression, so I believe it should be covered